### PR TITLE
[Bugfix] fix rewrite bug after insert new partition data

### LINKED ISSUE #19817: [materialized view] rewrite failed after insert data
### Steps to reproduce the behavior (Required)
````sql
CREATE TABLE t1 (
                    k1 INT,
                    v1 INT,
                    v2 INT)
                DUPLICATE KEY(k1)
                PARTITION BY RANGE(`k1`)
                (
                PARTITION `p1` VALUES LESS THAN ('2'),
                PARTITION `p2` VALUES LESS THAN ('3'),
                PARTITION `p3` VALUES LESS THAN ('4'),
                PARTITION `p4` VALUES LESS THAN ('5'),
                PARTITION `p5` VALUES LESS THAN ('6'),
                PARTITION `p6` VALUES LESS THAN ('7')
                )
                DISTRIBUTED BY HASH(k1);
                insert into t1 values (1,1,1),(1,1,2),(1,1,3),(1,2,1),(1,2,2),(1,2,3),(1,3,1),(1,3,2),(1,3,3)
                                     ,(2,1,1),(2,1,2),(2,1,3),(2,2,1),(2,2,2),(2,2,3),(2,3,1),(2,3,2),(2,3,3)
                                     ,(3,1,1),(3,1,2),(3,1,3),(3,2,1),(3,2,2),(3,2,3),(3,3,1),(3,3,2),(3,3,3);
CREATE MATERIALIZED VIEW mv2 PARTITION BY k1 DISTRIBUTED BY HASH(k1) BUCKETS 10
                 REFRESH ASYNC AS SELECT k1, count(v1) as count_k2, sum(v2) as sum_k3 from t1 group by k1;
alter materialized view mv2 rename mv_after;
SHOW MATERIALIZED VIEWS WHERE NAME LIKE 'mv_after';

SELECT sleep(5);
explain SELECT k1, count(v1), sum(v2) from t1 group by k1;
insert into t1 values (4,1,1);
SELECT sleep(5);
explain SELECT k1, count(v1), sum(v2) from t1 group by k1;
````

### Expected behavior (Required)
`TABLE: mv2`
### Real behavior (Required)
`TABLE: t1`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
`BRANCH-3.0-RELEASE-084e91b`

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19817 

## Problem Summary(Required) ：

Fix rewrite failure after inserting new partition data. The bug reason is that the mv plan is cached, but the plan may change after ingestion with new partitions data. And the logic of partition predicate calculation for mv rewrite depends on scan node's selected partition id, which may change after ingestion. So it leads to invalid compensation predicate in mv rewrite, which caused an invalid rewritten plan. Fix it by:
1. disable partition prune rules during compile mv plan, which will keep partition predicates in mv plan, to avoid the problem of caching plan
2. remove mv partition predicate compensation logic because it is not necessary after step 1. and it also fixed the problem of redundant partition predicates after partition prune of mv.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java [hunk 1]
```diff
@@ -230,7 +229,7 @@ private LogicalOlapScanOperator createScanMvOperator(MaterializationContext mvCo
         // Construct partition/distribution key column refs to filter conjunctions which need to retain.
         Set<String> mvPruneKeyColNames = Sets.newHashSet();
         distributedColumns.stream().forEach(distKey -> mvPruneKeyColNames.add(distKey.getName()));
-        mv.getPartitionNames().stream().forEach(partName -> mvPruneKeyColNames.add(partName));
+        mv.getPartitionColumnNames().stream().forEach(partName -> mvPruneKeyColNames.add(partName));
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java:232]: Why change it from `getPartitionNames` to `getPartitionColumnNames()` ?

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java:232]: the logic is to find the partition related predicates, should use partition column names to find them, not partition names.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java [hunk 2]
```diff
@@ -251,13 +250,6 @@ private LogicalOlapScanOperator createScanMvOperator(MaterializationContext mvCo
                 mvConjuncts.add(conj);
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java:250]: Do we still need `mvConjuncts ` anymore if no partition prunes?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java [hunk 3]
```diff
@@ -334,6 +334,7 @@ public static Pair<OptExpression, LogicalPlan> getRuleOptimizedLogicalPlan(Strin
                 new RelationTransformer(columnRefFactory, connectContext).transformWithSelectLimit(query);
         // optimize the sql by rule and disable rule based materialized view rewrite
         OptimizerConfig optimizerConfig = new OptimizerConfig(OptimizerConfig.OptimizerAlgorithm.RULE_BASED);
+        optimizerConfig.disableRuleSet(RuleSetType.PARTITION_PRUNE);
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java:337]: This only affects MV's optimination? what about Query's `compensatePartitionPredicate`, query still need `compensatePartitionPredicate`?

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java:337]: yes, query need to add total range to calculate compensation predicates

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java (hunk 2 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java:250]: remove it here, move it to rewriter.

**Review by murphyatwork — APPROVED** [review]

**Review by Dshadowzh — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.5 branch-3.0

**Pull request merged**